components/tenant-mgt/org.wso2.carbon.tenant.mgt/src/main/java/org/wso2/carbon/tenant/mgt/util/TenantMgtUtil.java

```diff
@@ -348,6 +351,62 @@ public static Tenant initializeTenant(TenantInfoBean tenantInfoBean) {
         return tenant;
     }
 
+    public static void updateTenantPassword(TenantInfoBean tenantInfoBean, String adminPassword)
+            throws TenantMgtException {
+
+        if (StringUtils.isBlank(adminPassword)) {
+            return;
+        }
+        UserStoreManager userStoreManager;
+        try {
+            userStoreManager = getUserStoreManager(tenantInfoBean.getTenantId());
+        } catch (org.wso2.carbon.user.api.UserStoreException e) {
+            String msg = "Error in getting the user store manager for tenant, tenant domain: " +
+                    tenantInfoBean.getTenantDomain() + ".";
+            if (log.isDebugEnabled()) {
+                log.debug(msg, e);
+            }
+            throw new TenantManagementServerException(msg, e);
+        }
+        try {
+            if (!userStoreManager.isReadOnly()) {
+                try {
+                    userStoreManager.updateCredentialByAdmin(tenantInfoBean.getAdmin(), adminPassword);
+                } catch (UserStoreException e) {
+                    String msg = String.format("Error in changing the tenant admin password for tenant domain: %s. %s",
+                            tenantInfoBean.getTenantDomain(), e.getMessage());
+                    if (log.isDebugEnabled()) {
+                        log.debug(msg, e);
+                    }
+                    throw new TenantManagementClientException(null, msg);
+                }
+            }
+        } catch (UserStoreException e) {
+            throw new TenantManagementServerException("Error in updating the tenant admin password.", e);
+        }
+    }
+
+    private static UserStoreManager getUserStoreManager(int tenantId) throws
+            org.wso2.carbon.user.api.UserStoreException {
+
+    RealmService realmService = TenantMgtServiceComponent.getRealmService();
+    if (realmService == null) {
+        throw new org.wso2.carbon.user.api.UserStoreException("RealmService is null");
```

Comment:
Shall we add these errors as proper error codes and messages from the error class constants just like 
```
ERROR_CODE_PARTIALLY_CREATED_OR_UPDATED.getCode(),
                    ERROR_CODE_PARTIALLY_CREATED_OR_UPDATED.getMessage()
```

Reply:
This error is handled respectively at their usages with `TenantManagementServerException` with different error codes for different use case